synapse/storage/databases/main/registration.py

```diff
@@ -224,22 +224,29 @@ async def set_renewal_token_for_user(
             desc="set_renewal_token_for_user",
         )
 
-    async def get_user_from_renewal_token(self, renewal_token: str) -> str:
-        """Get a user ID from a renewal token.
+    async def get_user_from_renewal_token(
+        self, renewal_token: str
+    ) -> Tuple[str, bool, int]:
+        """Get a user ID and renewal status from a renewal token.
 
         Args:
             renewal_token: The renewal token to perform the lookup with.
 
         Returns:
-            The ID of the user to which the token belongs.
+            A tuple of containing the following values:
+                * The ID of a user to which the token belongs.
+                * Whether an email has been sent to the user with the token, and
+                    that they haven't renewed their account with it yet.
```

Comment:
```suggestion
                * The ID of a user to which the token belongs.
                * Whether an email has been sent to the user with the token, and
                    that they haven't renewed their account with it yet.
                * An int representing the user's expiry timestamp as milliseconds since the
                    epoch, or 0 if the token was invalid.
```